src/main.js

```diff
@@ -251,6 +251,18 @@ const mixin = {
             elem.click();
             elem.remove();
         },
+        appendQueryToUrl(key, value) {
+            let searchParams = new URLSearchParams(window.location.search);
+            searchParams.set(key, value);
+            let newurl =
+                window.location.protocol +
+                "//" +
+                window.location.host +
+                window.location.pathname +
+                "?" +
+                searchParams.toString();
+            window.history.pushState({ path: newurl }, "", newurl);
```

Comment:
Rather than pushState, which creates a history entry in the browser, why not replaceState? I think this would result in a cleaner browser history.

Todo to self: Refactor all these helper functions to JS files

Reply:
I also noticed that this can be done a lot simpler: https://developer.mozilla.org/en-US/docs/Web/API/History/pushState#change_a_query_parameter